fvm/transactionSequenceNum.go

```diff
@@ -72,8 +75,11 @@ func (c *TransactionSequenceNumberChecker) checkAndIncrementSequenceNumber(
 
 	_, err = accounts.SetPublicKey(proposalKey.Address, proposalKey.KeyIndex, accountKey)
 	if err != nil {
-		childState.View().DropDelta()
+		sth.DisableAllLimitEnforcements()
+		_ = sth.RestartNestedTransaction(nestedTxnId)
```

Comment:
Why the extra DisableAllLimitEnforcements here?

Also... A restart is not needed here. Since this is the end of the transaction. Maybe just call commit.

Reply:
If we don't disable the limit, restart may fail since restart merges aborted nested transactions internally.

I'll add a comment